### PR TITLE
Implement cloud auth, session, and session commands

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
This PR adds three new commands:
- `tuist cloud auth`: To authenticate on the cloud server.
- `tuist cloud session`: To print the local user session.
- `tuist cloud session`: To remove the user local session.

### CODE REVIEW

File: Sources/TuistKit/Commands/Cloud/CloudLogoutCommand.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import SPMUtility
+import TuistCore
+import TuistSigning
+import TuistSupport
+
+class CloudLogoutCommand: NSObject, Command {
+    // MARK: - Attributes
+
+    static let command = "logout"
+    static let overview = "Removes any existing session to authenticate on the server with he URL defined in the Config.swift file."
```
Comment (andreacipriani) [inline Sources/TuistKit/Commands/Cloud/CloudLogoutCommand.swift:12]:
```suggestion
    static let overview = "Removes any existing session to authenticate on the server with the URL defined in the Config.swift file."
```

File: Sources/TuistKit/Commands/Cloud/CloudSessionCommand.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import SPMUtility
+import TuistCore
+import TuistSigning
+import TuistSupport
+
+class CloudSessionCommand: NSObject, Command {
+    // MARK: - Attributes
+
+    static let command = "session"
+    static let overview = "Prints any existing session to authenticate on the server with he URL defined in the Config.swift file."
```
Comment (andreacipriani) [inline Sources/TuistKit/Commands/Cloud/CloudSessionCommand.swift:12]:
```suggestion
    static let overview = "Prints any existing session to authenticate on the server with he URL defined in the Config.swift file."
```
```suggestion
    static let overview = "Prints any existing session to authenticate on the server with the URL defined in the Config.swift file."
```

File: Sources/TuistKit/Services/Cloud/CloudAuthService.swift [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    init(cloudSessionController: CloudSessionControlling,
+         generatorModelLoader: GeneratorModelLoading) {
+        self.cloudSessionController = cloudSessionController
+        self.generatorModelLoader = generatorModelLoader
+    }
+
+    // MARK: - CloudAuthServicing
+
+    func authenticate() throws {
+        let path = FileHandler.shared.currentPath
```
Comment (andreacipriani) [inline Sources/TuistKit/Services/Cloud/CloudAuthService.swift:61]: does this mean that the commands only work if you run them from the same path were the Config file is?

Comment (pepibumur) [reply Sources/TuistKit/Services/Cloud/CloudAuthService.swift:61]: Not really. The config loader traverses the up the folders hierarchy until it finds a `Tuist/Config.swift` file.

Comment (andreacipriani) [reply Sources/TuistKit/Services/Cloud/CloudAuthService.swift:61]: ok that's cool, I just didn't get why we're using `currentPath` as the path

**Pull request merged**